### PR TITLE
examples: add image-blur example

### PR DESCRIPTION
**Pull request by lucasromanosantos**
hey! this is my first zig thing so feel free to provide the most feedback you can.

`filter_size`, `iterations` and `block_dimension` are hardcoded for now as the UI is being developed.

- [x] By selecting this checkbox, I agree to license my contributions to this project under the license(s) described in the LICENSE file, and I have the right to do so or have received permission to do so by an employer or client I am producing work for whom has this right.

### CODE REVIEW

File: examples/image-blur/main.zig [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+
+pub fn init(app: *App, core: *mach.Core) !void {
+    const queue = core.device.getQueue();
+
+    try core.setOptions(.{
+        .size_min = .{ .width = 20, .height = 20 },
+    });
+
+    const blur_shader_module = core.device.createShaderModule(&.{
+        .label = "my blur shader module",
```
Comment (slimsag) [inline examples/image-blur/main.zig:38]:
```suggestion
        .label = "blur shader",
```

File: examples/image-blur/main.zig [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+        .compute = gpu.ProgrammableStageDescriptor{
+            .module = blur_shader_module,
+            .entry_point = "main",
+        },
+    };
+
+    const blur_pipeline = core.device.createComputePipeline(&blur_pipeline_descriptor);
+
+    const fullscreen_quad_vs_module = core.device.createShaderModule(&.{
+        .label = "my fullscreen quad vertex shader",
```
Comment (slimsag) [inline examples/image-blur/main.zig:52]:
```suggestion
        .label = "fullscreen quad vertex shader",
```

File: examples/image-blur/main.zig [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+        .code = .{ .wgsl = @embedFile("fullscreen_textured_quad.wgsl") },
+    });
+
+    const fullscreen_quad_fs_module = core.device.createShaderModule(&.{
+        .label = "my fullscreen quad fragment shader",
```
Comment (slimsag) [inline examples/image-blur/main.zig:57]:
```suggestion
        .label = "fullscreen quad fragment shader",
```

File: examples/image-blur/main.zig [hunk 4]
```diff
@@ -0,0 +1,290 @@
...
+        .Rgb24 => |pixels| {
+            const data = try rgb24ToRgba32(core.allocator, pixels);
+            defer data.deinit(core.allocator);
+            queue.writeTexture(&.{ .texture = cube_texture }, &data_layout, &img_size, zigimg.color.Rgba32, data.Rgba32);
+        },
+        else => @panic("unsupported image color format"),
+    }
+
+    var textures: [2]gpu.Texture = undefined;
+    for ([_]i32{ 0, 1 }) |_, i| {
```
Comment (slimsag) [inline examples/image-blur/main.zig:135]: Change this to `for (textures) |_, i| {`, then it's not possible to work on the wrong amount of items.

File: examples/image-blur/main.zig [hunk 5]
```diff
@@ -0,0 +1,290 @@
...
+                .texture_binding = true,
+                .copy_dst = true,
+            },
+        });
+    }
+
+    // the shader blurs the input texture in one direction,
+    // depending on whether flip value is 0 or 1
+    var flip: [2]gpu.Buffer = undefined;
+    for ([_]u32{ 0, 1 }) |v, i| {
```
Comment (slimsag) [inline examples/image-blur/main.zig:150]: same here

**Comment by slimsag** [comment]
Actually the quality here looks great aside from the few suggestions I left. This is awesome, nice work!

**Comment by lucasromanosantos** [comment]
Done! Thanks @slimsag 😄

**Review by slimsag — APPROVED** [review]

**Comment by slimsag** [comment]
Nice work! This is cool :)

Alright if I add you to the contributors list at https://machengine.org/gpu/ as well?

**Pull request merged**